apps/condo/domains/user/schema/UserExternalIdentity.js

```diff
@@ -0,0 +1,86 @@
+const { Text, Select, Relationship } = require('@keystonejs/fields')
+const { Json } = require('@open-condo/keystone/fields')
+const { GQLListSchema, getById } = require('@open-condo/keystone/schema')
+const { historical, versioned, uuided, tracked, softDeleted, dvAndSender } = require('@open-condo/keystone/plugins')
+
+const { IDP_TYPES, RESIDENT } = require('@condo/domains/user/constants/common')
+
+const access = require('@condo/domains/user/access/UserExternalIdentity')
+
+const UserExternalIdentity = new GQLListSchema('UserExternalIdentity', {
+    schemaDoc: 'Individual user external identity link. Used primarily for authorization and linking to external idp purposes. Think of `User` as a technical entity, not a business actor.',
+    fields: {
+        user: {
+            schemaDoc: 'Link to user',
+            type: Relationship,
+            ref: 'User',
+            isRequired: true,
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.PROTECT' },
+            hooks: {
+                validateInput: async ({ operation, resolvedData, fieldPath, addFieldValidationError }) => {
+                    if (operation === 'create') {
+                        const userId = get(resolvedData, fieldPath)
+                        if (!userId) {
+                            addFieldValidationError('No user specified')
+                            return
+                        }
+                        const user = await getById('User', userId)
+                        const userType = get(user, 'type')
+                        if ( userType !== RESIDENT) {
+                            addFieldValidationError('Cannot create UserExternalIdentity to non-resident type of user')
+                        }
+                    }
+                },
+            },
+        },
+        identityId: {
+            schemaDoc: 'External identity id. The value of this field should be populated from an external identity provider',
+            type: Text,
+        },
+        identityType: {
+            schemaDoc: 'The type of external identity that was a source for this link',
+            type: Select,
+            dataType: 'enum',
+            options: IDP_TYPES,
+            isRequired: true,
+        },
+        meta: {
+            schemaDoc: 'External identity provider user metadata',
+            type: Json,
+        },
+    },
+    kmigratorOptions: {
+        constraints: [
+            {
+                type: 'models.UniqueConstraint',
+                fields: ['identityId', 'identityType'],
+                name: 'unique_identityid_and_identitytype',
+            },
+        ],
+    },
+    plugins: [
+        uuided(),
+        versioned(),
+        tracked(),
+        softDeleted(),
+        dvAndSender(),
+        historical(),
+    ],
+    access: {
+        read: access.canReadUserExternalIntegrations,
+        create: false,
```

Comment:
You need to use can manage here. Because you need to have GQL create input

Reply:
I supposed to use RegisterUserExternalIdentityService to create UserExternalIdenty. Now it doesn't have specific business logic in here. But once we will integrate it with sbbol/sberId this service will be a place for business logic.

Reply:
Usually, if you don't have any specific BL you really don't need to add a custom mutation